### PR TITLE
fix(editor): preserve safe rich Markdown literals

### PR DESCRIPTION
**Pull request by nwparker**
| | Files | Added | Deleted | Net |
| :--- | ---: | ---: | ---: | ---: |
| Test | 1 | $\color{#1a7f37}{\Huge{\mathbf{+}}}$​179 | 0 | $\color{#1a7f37}{\Huge{\mathbf{+}}}$​179 |
| Prod | 1 | $\color{#1a7f37}{\Huge{\mathbf{+}}}$​27 | $\color{#cf222e}{\Huge{\mathbf{−}}}$​6 | $\color{#1a7f37}{\Huge{\mathbf{+}}}$​21 |

## ELI5

Editing rich Markdown keeps safe underscores and nested callouts readable in the saved source.

## What Changed

Extend the existing literal-source preservation guard to Unicode intraword underscore runs and nested quote/list blocks. Each optional unescape is accepted only after reparsing produces the same editor node. Keep reference-like text conservative, bound the newly added validation to 50,000 code units per block and reuse the existing node cache. Preserve the existing large paragraph/heading bracket path, so long wiki-link prose keeps its established source fidelity.

## Why

Removing escapes by regex alone can turn literal `__init__` or `___init___` into emphasis on reopening. The existing parse-and-node-equality guard proves the saved syntax retains its meaning. Adapted from @nvimq #22123; attribution is preserved.

## Linked Issue

Related to #22014 and #22123. This covers safe underscores and nested callouts; it does not claim to preserve every source-formatting detail.

## Visual Proof

Real hidden macOS Orca save/reopen flow in a folder workspace. Before/after saved source, baseline `b76bc79d73d` and the same preservation change from the prior validation pass. Cropped to exclude local paths. Equality was checked from saved Source-to-Rich state to reopened state, not against an unsaved pre-serialization snapshot.

## Testing

- [x] I manually tested these changes locally
- [x] Automated tests added/updated

Current-base checks: all 368 rich-Markdown tests pass (2 skipped), including the existing large malformed-link paragraph regression; renderer types and changed-code quality pass. Prior full focused packet: 141 tests and hidden save/reopen checks passed; 11 regressions failed on baseline.

## Review

Reviewed Unicode and escaped-backslash handling, delimiter runs, reference-definition interactions across blocks, raw HTML/code source ownership, bounded reparse work and cache invalidation. Unsafe or unproven changes retain the canonical escaped output.

## Agent skill upstream boundary

- [x] Not applicable; no agent skill source changes.

## Notes

Blank-line/final-newline normalization, other escape characters, definition-like/unsafe blocks and oversized blocks outside the existing paragraph/heading bracket path remain unchanged. The size cap and cache tests are not latency benchmarks. Local validation did not include native Windows/Linux/WSL, live SSH, mobile, the full suite or packaging. Required PR Checks and packaging passed in CI; the completed current-head Pullfrog review also reports 1,703 editor tests passed with two skipped. An additional requested Pullfrog review is still running. Host-aware save routing and remote wire behavior are unchanged.

## Checklist

- [x] This PR is small and focused
- [x] I explained what changed and why
- [x] Before/after screenshots attached
- [x] Self-reviewed for correctness, security, and performance
- [x] Cross-platform, SSH/remote, and path/shortcut impact considered
- [x] Required PR Checks verification, static analysis, types, tests and packaging pass; focused local checks are listed above

### CODE REVIEW

**Comment by nwparker** [comment]
@coderabbitai review

**Comment by nwparker** [comment]
@pullfrog Please review the current head, 7a1ddd3973209b1738ae7e77a8d199daa22c596e. The initial review is still running against 4f3b095245d2. Since then, CI identified the existing large malformed-link paragraph regression; the size limit now preserves the existing paragraph/heading bracket path. All 368 rich-Markdown tests pass locally (2 skipped), along with renderer types and changed-code quality.

File: src/renderer/src/components/editor/rich-markdown-literal-serialization.ts [hunk 1]
```diff
@@ -3,9 +3,21 @@ import type { Node as ProseMirrorNode } from '@tiptap/pm/model'
 import { encodeRawMarkdownHtmlForRichEditor } from './raw-markdown-html'
 import type { RichMarkdownEditorCodec } from './rich-markdown-source-transport'
 
+const MAX_LITERAL_BLOCK_CODE_UNITS = 50_000
```
[the remark this answers is not in the document]
Comment (nwparker) [reply src/renderer/src/components/editor/rich-markdown-literal-serialization.ts:6]: Fixed in 7a1ddd3973209b1738ae7e77a8d199daa22c596e, before this delayed review arrived. Existing paragraph/heading bracket preservation bypasses the new size cap; only the expanded optional-validation paths retain the bound. The original large malformed-link test is unchanged and passes. All 368 rich-Markdown tests pass locally (2 skipped), and the current PR Checks run 35825116602 is green, including the previously failing shard. The PR description explicitly documents the remaining bound on expanded cases and uses Related to #22014 rather than claiming complete source-format preservation.

**Comment by nwparker** [comment]
Follow-up to the review of `4f3b095245d2`:

- The large-paragraph regression was already fixed in `7a1ddd397320`; the original exact-source test remains unchanged and passes, as do all 368 rich-Markdown tests (2 skipped). Current PR Checks is green.
- The broad reference-like guard is deliberately conservative in this scoped change. Definitions can affect other blocks, including nested containers; the tests cover those interactions. Narrowing it with another Markdown-shape regex would need a separate grammar/edge-case review. The description lists definition-like text among the remaining formatting limits.
- The delimiter/backslash cases assert the intended safety property: reopening must preserve the complete editor tree. Literal `__init__` and `_word_` cannot simply be emitted unescaped, because that would turn them into emphasis. Exact output is pinned for intraword identifiers in the separate preservation cases; the delimiter cases allow safe canonical escaping while rejecting semantic corruption.
